gtdynamics/utils/Initializer.cpp

```diff
@@ -145,8 +144,11 @@ Values Initializer::InitializeSolutionInterpolation(
     InsertTwist(&init_vals, link->id(), t, gtsam::Z_6x1);
     init_vals = robot.forwardKinematics(init_vals, t, link_name);
 
-    for (auto&& kvp : ZeroValues(robot, t, gaussian_noise, contact_points)) {
-      init_vals.tryInsert(kvp.key, kvp.value);
+    Values zero_values = ZeroValues(robot, t, gaussian_noise, contact_points);
+    for (auto&& key : zero_values.keys()) {
```

Comment:
This works, but I think this is even better:
init_vals.insert(zero_values);

Reply:
So that won't work, because the code expects only insertion and not updates. If a key already exists, the expectation is to simply skip it.

This pattern is very weird to me (I didn't write this) so I am not trying to make too many changes.